pkg/resource/deploy/source_eval.go

```diff
@@ -1175,6 +1187,48 @@ func (rm *resmon) RegisterResource(ctx context.Context,
 	var result *RegisterResult
 
 	var outputDeps map[string]*pulumirpc.RegisterResourceResponse_PropertyDependencies
+
+	additionalSecretKeys := make([]resource.PropertyKey, 0, len(additionalSecretOutputs))
+	for _, name := range additionalSecretOutputs {
+		additionalSecretKeys = append(additionalSecretKeys, resource.PropertyKey(name))
+	}
+	var timeouts resource.CustomTimeouts
+	if customTimeouts != nil {
+		if customTimeouts.Create != "" {
+			seconds, err := generateTimeoutInSeconds(customTimeouts.Create)
+			if err != nil {
+				return nil, err
+			}
+			timeouts.Create = seconds
+		}
+		if customTimeouts.Delete != "" {
+			seconds, err := generateTimeoutInSeconds(customTimeouts.Delete)
+			if err != nil {
+				return nil, err
+			}
+			timeouts.Delete = seconds
+		}
+		if customTimeouts.Update != "" {
+			seconds, err := generateTimeoutInSeconds(customTimeouts.Update)
+			if err != nil {
+				return nil, err
+			}
+			timeouts.Update = seconds
+		}
+	}
+
+	goal := resource.NewGoal(t, name, custom, props, parent, protect, dependencies,
+		providerRef.String(), nil, propertyDependencies, deleteBeforeReplace, ignoreChanges,
+		additionalSecretKeys, aliases, id, &timeouts, replaceOnChanges, retainOnDelete, deletedWith)
+
+	if goal.Parent != "" {
+		parentGoal, ok := rm.resGoals[goal.Parent]
+		if ok {
+			goal = inheritFromParent(*goal, parentGoal)
+		}
+	}
```

Comment:
Logic moved higher up to create a Goal even when a resource is a remote resource.

Should I leave the code in there and only construct if it's not a remote resource? On further inspection, provider.Construct() assumes/expects the provider to call RegisterResource and may potentially be better, but I'd like a second opinion from the MLC experts.

Reply:
Yeah, I think this should be moved back to where it was.